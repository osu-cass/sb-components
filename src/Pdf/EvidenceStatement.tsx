import * as React from 'react';

export interface EvidenceStatementProps {
    statement: string;
}

export class EvidenceStatement extends React.Component<EvidenceStatementProps, {}> {
    render() {
        return (
            <div className="centered-table-container">
                <table className='item-data-table'>
                    <thead>
                        <tr>
                            <th>Evidence Statement</th>
                        </tr>
                    </thead>
                    <tbody>
                        <tr>
                            <td>{this.props.statement}</td>
                        </tr>
                    </tbody>
                </table>
            </div>
        );
    }
}