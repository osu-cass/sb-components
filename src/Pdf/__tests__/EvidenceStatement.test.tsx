import * as React from 'react';
import * as ReactDOM from 'react-dom';
import { EvidenceStatement } from '../EvidenceStatement';
import { shallow, mount, render } from 'enzyme';

describe("EvidenceStatement", () => {
    it("matches snapshot", () => {
        const wrapper = shallow(<EvidenceStatement statement="" />)
        expect(wrapper).toMatchSnapshot();
    })
})
