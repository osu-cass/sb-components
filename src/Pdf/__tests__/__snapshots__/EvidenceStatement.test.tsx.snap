// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EvidenceStatement matches snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <EvidenceStatement
    statement=""
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <table
        className="item-data-table"
      >
        <thead>
          <tr>
            <th>
              Evidence Statement
            </th>
          </tr>
        </thead>
        <tbody>
          <tr>
            <td>
              
            </td>
          </tr>
        </tbody>
      </table>,
      "className": "centered-table-container",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <thead>
            <tr>
              <th>
                Evidence Statement
              </th>
            </tr>
          </thead>,
          <tbody>
            <tr>
              <td>
                
              </td>
            </tr>
          </tbody>,
        ],
        "className": "item-data-table",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <tr>
              <th>
                Evidence Statement
              </th>
            </tr>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <th>
                Evidence Statement
              </th>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Evidence Statement",
              },
              "ref": null,
              "rendered": "Evidence Statement",
              "type": "th",
            },
            "type": "tr",
          },
          "type": "thead",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <tr>
              <td>
                
              </td>
            </tr>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <td>
                
              </td>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "",
              },
              "ref": null,
              "rendered": "",
              "type": "td",
            },
            "type": "tr",
          },
          "type": "tbody",
        },
      ],
      "type": "table",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <table
          className="item-data-table"
        >
          <thead>
            <tr>
              <th>
                Evidence Statement
              </th>
            </tr>
          </thead>
          <tbody>
            <tr>
              <td>
                
              </td>
            </tr>
          </tbody>
        </table>,
        "className": "centered-table-container",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <thead>
              <tr>
                <th>
                  Evidence Statement
                </th>
              </tr>
            </thead>,
            <tbody>
              <tr>
                <td>
                  
                </td>
              </tr>
            </tbody>,
          ],
          "className": "item-data-table",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <tr>
                <th>
                  Evidence Statement
                </th>
              </tr>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <th>
                  Evidence Statement
                </th>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Evidence Statement",
                },
                "ref": null,
                "rendered": "Evidence Statement",
                "type": "th",
              },
              "type": "tr",
            },
            "type": "thead",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <tr>
                <td>
                  
                </td>
              </tr>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <td>
                  
                </td>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "",
                },
                "ref": null,
                "rendered": "",
                "type": "td",
              },
              "type": "tr",
            },
            "type": "tbody",
          },
        ],
        "type": "table",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
